neo/rawio/spikegadgetsrawio.py

```diff
@@ -0,0 +1,247 @@
+        for device in hconf:
+            stream_id = device.attrib['name']
+            num_bytes = int(device.attrib['numBytes'])
+            stream_bytes[stream_id] = packet_size
+            packet_size += num_bytes
+        
+        # timesteamps 4 uint32
+        self._timestamp_byte = packet_size
+        packet_size += 4
+        
+        packet_size += 2 * num_ephy_channels
+
+        # read the binary part lazily
+        raw_memmap = np.memmap(self.filename, mode='r', offset=header_size, dtype='<u1')
+
+        num_packet = raw_memmap.size // packet_size
+        raw_memmap = raw_memmap[:num_packet*packet_size]
+        self._raw_memmap = raw_memmap.reshape(-1, packet_size)
+
+        # create signal channels
+        stream_ids = []
+        signal_streams = []
+        signal_channels = []
+        
+        # walk in xml device and keep only "analog" one
+        self._mask_channels_bytes  = {}
+        for device in hconf:
+            stream_id = device.attrib['name']
+            for channel in device:
+
+                if 'interleavedDataIDByte' in channel.attrib:
+                    # TODO deal with "headstageSensor" wich have interleaved
+                    continue
+
+                if channel.attrib['dataType'] == 'analog':
+                    
+                    if stream_id not in stream_ids:
+                        stream_ids.append(stream_id)
+                        stream_name = stream_id
+                        signal_streams.append((stream_name, stream_id))
+                        self._mask_channels_bytes[stream_id] = []
+                    
+                    name = channel.attrib['id']
+                    chan_id = channel.attrib['id']
+                    dtype = 'int16'
+                    units = 'uV' # TODO check where is the info
+                    gain = 1. # TODO check where is the info
+                    offset = 0.
+                    signal_channels.append((name, chan_id, self._sampling_rate, 'int16',
+                                         units, gain, offset, stream_id))
+                    
+                    num_bytes = stream_bytes[stream_id] + int(channel.attrib['startByte'])
+                    chan_mask = np.zeros(packet_size, dtype='bool')
+                    chan_mask[num_bytes] = True
+                    chan_mask[num_bytes+1] = True
+                    self._mask_channels_bytes[stream_id].append(chan_mask)
+        
+        if num_ephy_channels > 0:
+            stream_id = 'trodes'
+            stream_name = stream_id
+            signal_streams.append((stream_name, stream_id))
+            self._mask_channels_bytes[stream_id] = []
+            
+            chan_ind = 0
+            for trode in sconf:
+                for  schan in trode:
+                    name = 'trode' + trode.attrib['id'] + 'chan' + schan.attrib['hwChan']
+                    chan_id = schan.attrib['hwChan']
+                    units = 'uV' # TODO check where is the info
```

Comment:
same here...?

Reply:
at the moment units won't be handle in this IOs